### PR TITLE
add Packer configuration to automate creating DO image

### PR DESCRIPTION
**Pull request by Mohammed90**
The scripts `90-cleanup.sh` and `99-img_check.sh` are from [DO's own repo](https://github.com/digitalocean/marketplace-partners/tree/master/scripts).

The droplet is hardened with these roles:
- [dev-sec.os-hardening](https://github.com/dev-sec/ansible-os-hardening)
- [dev-sec.ssh-hardening](https://github.com/dev-sec/ansible-ssh-hardening)

### CODE REVIEW

File: packer/scripts/20-caddy.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    --gid caddy \
+    --create-home \
+    --home-dir /var/lib/caddy \
+    --shell /usr/sbin/nologin \
+    --comment "Caddy web server" \
+    caddy
+
+chown -R caddy:caddy /var/lib/caddy
+
+wget -O /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/v2.0.0-beta8/caddy2_beta8_linux_amd64
```
Comment (hairyhenderson) [inline packer/scripts/20-caddy.sh]: Not a huge deal, but it'd be good to extract the version number to a variable, which could be fed in through the provisioner (using the [`environment_vars`](https://www.packer.io/docs/provisioners/shell.html#environment_vars) key).

Also to limit terminal output it's probably useful to do `wget -qO` instead:

```suggestion
wget -qO /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/v2.0.0-beta8/caddy2_beta8_linux_amd64
```

Comment (Mohammed90) [reply packer/scripts/20-caddy.sh]: Done

File: packer/scripts/20-caddy.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    --create-home \
+    --home-dir /var/lib/caddy \
+    --shell /usr/sbin/nologin \
+    --comment "Caddy web server" \
+    caddy
+
+chown -R caddy:caddy /var/lib/caddy
+
+VERSION_SLUG=$(echo $CADDY_VERSION | sed 's/-/_/g')
+wget -qO /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/v${CADDY_VERSION}/caddy${VERSION_SLUG}_linux_amd64
```
Comment (Mohammed90) [inline packer/scripts/20-caddy.sh]: @mholt , can we standardize how the version and binary name will look like? The `sed` operation here turns `v2.0.0-beta9` to `v2.0.0_beta9`. That's not how you've named the binaries in releases of v2. The last one was `caddy2_beta9_linux_amd64`. To go from `v2.0.0-beta9` to `caddy2_beta9` I can do the following line:

`VERSION_SLUG=$(echo $CADDY_VERSION | sed 's/-/_/g' | sed 's/\.[0-9]*\.[0-9]*//g')`

But you'd lose the point releases in the binary name. Let's agree on a format to better standardize our scripts along with binary names.

Comment (mholt) [reply packer/scripts/20-caddy.sh]: Sure. Right now I'm manually building and naming each binary, but once the process is automated (probably sometime around the release candidates), they'll be consistent.

Would it just be easier to use the version tag, unchanged, in the name?

Also, note that the automated builds will probably be zip/tar files archives.

Comment (Mohammed90) [reply packer/scripts/20-caddy.sh]: So something like this

```
ver=$(git describe --abbrev=0)
geese=("linux" "windows" "macos")
goarches=("amd64" "x86")

for goos in "${geese[@]}"
do
    for goarch in "${goarches[@]}"
    do
        GOOS=$goos GOARCH=$goarch go build -o caddy${ver}_${os}_${arch} -trimpath main.go
    done
done
```
?

We can adjust this file later when we start having automated zip/tar archives.

**Pull request merged**